api/envoy/annotations/deprecation.proto

```diff
@@ -0,0 +1,15 @@
+syntax = "proto3";
+
+package envoy.annotations;
+
+import "google/protobuf/descriptor.proto";
+
+// Magic number in this file derived from top 28bit of SHA256 digest of "envoy.annotation.do_not_use"
+extend google.protobuf.FieldOptions {
```

Comment:
One other interesting thing here is we're getting awfully close to being able to either encode the runtime key here as an annotation, or via reflection to being able to derive it. Not sure where the status quo there is today. Something worth thinking about, even if it's beyond the scope of this PR.